From ec8cdef9da8e86b286500313978d922a1430a059 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Thu, 5 Sep 2013 20:15:37 +0200
Subject: [PATCH] Fix build with gcc-4.8.1

:Release Notes:
Similar fix as:
  commit 85121b449d79f951fa24f3c5d50b1214ccad79e8
  Author: Igor Kuksyuk <ikuksyuk@luxoft.com>
  Date:   Tue Jul 30 18:26:53 2013 +0300

    Fix build issue with gcc-4.8.1

:Detailed Notes:
Replaced auto_ptr usage on boost::scoped_ptr if compiled with option "-std=c++11".
Fixes following build error when building with gcc-4.8*:
[ 47%] In file included from db8/2.0.0-80-r14/git/inc/db/MojDbShardEngine.h:27:0,
                 from db8/2.0.0-80-r14/git/inc/db/MojDb.h:31,
                 from db8/2.0.0-80-r14/git/test/db-media/MediaTest.cpp:30:
/usr/include/c++/backward/auto_ptr.h:293:12: error: '_Lock_policy' has not been declared
   template<_Lock_policy _Lp>
            ^
/usr/include/c++/backward/auto_ptr.h:296:5: error: '__shared_count' does not name a type
     __shared_count<_Lp>::__shared_count(std::auto_ptr<_Tp>&& __r)
     ^
/usr/include/c++/backward/auto_ptr.h:300:26: error: '_Lock_policy' has not been declared
   template<typename _Tp, _Lock_policy _Lp>
                          ^
/usr/include/c++/backward/auto_ptr.h:303:5: error: '__shared_ptr' does not name a type
     __shared_ptr<_Tp, _Lp>::__shared_ptr(std::auto_ptr<_Tp1>&& __r)
     ^
/usr/include/c++/backward/auto_ptr.h:316:5: error: 'shared_ptr' does not name a type
     shared_ptr<_Tp>::shared_ptr(std::auto_ptr<_Tp1>&& __r)
     ^
/usr/include/c++/backward/auto_ptr.h:322:5: error: 'unique_ptr' does not name a type
     unique_ptr<_Tp, _Dp>::unique_ptr(auto_ptr<_Up>&& __u) noexcept
     ^

:Testing Performed:
Only build tested

:QA Notes:

:Issues Addressed:
[GF-10391] db8 build fails with gcc-4.8.1

Open-webOS-DCO-1.0-Signed-off-by: Martin Jansa <martin.jansa@lge.com>
Change-Id: I7a6e0ff57071a3f823c6d665c614824c457e109a
---
 inc/db/MojDbShardEngine.h |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/inc/db/MojDbShardEngine.h b/inc/db/MojDbShardEngine.h
index e88dd24..f694ee2 100644
--- a/inc/db/MojDbShardEngine.h
+++ b/inc/db/MojDbShardEngine.h
@@ -24,7 +24,13 @@
 #include "core/MojErr.h"
 #include "core/MojString.h"
 #include "core/MojSignal.h"
-#include <auto_ptr.h>
+
+#ifdef __GXX_EXPERIMENTAL_CXX0X__
+    #include <boost/smart_ptr/scoped_ptr.hpp>
+#else
+    #include <auto_ptr.h>
+#endif
+
 #include <list>
 
 class MojDb;
-- 
1.7.10.4

